packages/expo-cellular/android/src/main/java/expo/modules/cellular/CellularModule.kt

```diff
@@ -0,0 +1,90 @@
+package expo.modules.cellular
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.net.sip.SipManager
+import android.os.Build
+import android.telephony.TelephonyManager
+import org.unimodules.core.ExportedModule
+import org.unimodules.core.ModuleRegistry
+import org.unimodules.core.ModuleRegistryDelegate
+import org.unimodules.core.Promise
+import org.unimodules.core.interfaces.ExpoMethod
+import org.unimodules.core.interfaces.RegistryLifecycleListener
+import java.util.*
+
+
+class CellularModule(
+  private val mContext: Context,
+  private val moduleRegistryDelegate: ModuleRegistryDelegate = ModuleRegistryDelegate(),
+) : ExportedModule(mContext), RegistryLifecycleListener {
+  override fun getName(): String = "ExpoCellular"
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    moduleRegistryDelegate.onCreate(moduleRegistry)
+  }
+
+  override fun getConstants(): HashMap<String, Any?> {
+    val constants = HashMap<String, Any?>()
+    constants["allowsVoip"] = SipManager.isVoipSupported(mContext)
+    val telephonyManager = mContext.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
```

Comment:
Android does not guarantee this value is non-null so we need to handle that case. There are some cases like Instant Apps where some system services are null.